src/renderers/shared/fiber/ReactFiberCommitWork.js

```diff
@@ -221,13 +222,27 @@ module.exports = function<T, P, I, TI, C>(config : HostConfig<T, P, I, TI, C>) {
     }
   }
 
-  function commitUnmount(current : Fiber) : void {
+  function commitUnmount(current : Fiber, safely : boolean) : void {
+    // Make sure we mark this deletion as completed so that we don't
+    // attempt to perform it again if one of the deletions fails, and
+    // error boundary tries to unmount this subtree again.
```

Comment:
It does look like you're tracking the currently unmounting Fiber anywhere so how do you know which error boundary to invoke if componentWillUnmount throws during a normal unmount?

In theory, since you need to track the currently unmounting Fiber you should be able to know which one failed. Since only one can fail, you can continue walking the tree from that point on. You'd know that anything after the failing node hasn't yet unmounted and anything before it has unmounted.